From ddd0d84ad7e086ebae4e797dfd1bebe3afdc95d6 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 25 Aug 2016 22:42:01 -0400
Subject: [PATCH] fix new.roots object is not iterable

------
  File "/usr/lib64/python2.7/site-packages/blivet/__init__.py", line 1904, in copy
    for root in new.roots:

TypeError: 'NoneType' object is not iterable
------

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 blivet/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/blivet/__init__.py b/blivet/__init__.py
index c1b35a3..5a6c8bc 100644
--- a/blivet/__init__.py
+++ b/blivet/__init__.py
@@ -1901,7 +1901,7 @@ class Blivet(object):
             p = partition.disk.format.partedDisk.getPartitionByPath(partition.path)
             partition.partedPartition = p
 
-        for root in new.roots:
+        for root in new.roots or []:
             root.swaps = [new.devicetree.getDeviceByID(d.id, hidden=True) for d in root.swaps]
             root.swaps = [s for s in root.swaps if s]
 
-- 
2.8.1

